controllers/evalhub/evalhub_controller.go

```
@@ -143,6 +146,29 @@ func (r *EvalHubReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
+			log.Error(err, "Failed to get instance namespace for placement check")
+			return RequeueWithError(err)
+		}
+		if _, isTenant := ns.Labels[tenantLabel]; isTenant {
+			log.Error(nil, "Multi-tenant EvalHub placed in tenant namespace", "namespace", instance.Namespace)
+			instance.SetStatus("Ready", invalidPlacementReason,
+				"A multi-tenant EvalHub cannot be deployed in a tenant namespace. "+
+					"Remove the evalhub.trustyai.opendatahub.io/tenant label from this namespace "+
+					"or set spec.tenancy: single.",
+				corev1.ConditionFalse)
+			instance.Status.Phase = "Error"
+			r.Status().Update(ctx, instance)
```

golangci-lint (2.12.2)

[error] Error return value of `(sigs.k8s.io/controller-runtime/pkg/client.SubResourceWriter).Update` is not checked

(errcheck)